Add a CMake option to select no std::filesystem (#350)
taocpp/PEGTL: CMakeLists.txt

target_compile_features(pegtl INTERFACE cxx_std_17)

option(PEGTL_USE_BOOST_FILESYSTEM "Override the auto-detection of std::filesystem and use Boost.Filesystem" OFF)

> option(PEGTL_NO_STD_FILESYSTEM "Don't link to any std::filesystem; file operations will not work" OFF)

Reviewer: ```suggestion
option(PEGTL_STD_FILESYSTEM "Link to any std::filesystem" ON)
```

I would suggest using positive option name instead of negative to avoid misconfiguration when using negative-negative value. Plus, it would enforce the current state, instead of "adding" a new meaning to the filesystem usage.

Author: Thank you for the feedback. I have updated the pull request accordingly to have a positive option and clearer documentation strings.
